### ISSUE TITLE
`Style/IfUnlessModifier` throws a false positive when the condition body raises an error

### ISSUE THREAD

**Issue by Hansenq** [rubocop/rubocop#9742]
Hello! I'm running into a false positive with the `Style/IfUnlessModifier` when the condition body raises an error: 

```ruby
# test.rb
super_long_variable_name = 12
if super_long_variable_name == 'my_other_super_long_string'
  raise 'this is my super long error message'
end
```

```bash
[22:33:01] %> rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:4:1: C: [Correctable] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
if super_long_variable_name == 'my_other_super_long_string'
^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

When corrected, it changes the line of code to the following: 
```ruby
super_long_variable_name = 12
raise 'this is my super long error message' if super_long_variable_name == 'my_other_super_long_string'
```

which is clearly incorrect, even at my line limit of 90 (set in `Layout/LineLength`). I've commented out all custom rules in my .rubocop.yml file and run this again, and I get the same error. Changing the condition body to any other line (such as an assignment or a method execution) fixes the issue, so I think it's only isolated to raising errors. 

--------

## Expected behavior
Rubocop should not raise this as a violation of the `Style/IfUnlessModifier` rule

## Actual behavior
Rubocop reports this block of code as violating the `Style/IfUnlessModifier` rule. Debug output below (DIR is my replacing of my computer's directories)
```
For /DIR: configuration from /DIR/.rubocop.yml
Inheriting configuration from /Users/user/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/prettier-1.5.5/rubocop.yml
Default configuration from /Users/user/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.13.0/config/default.yml
Inspecting 1 file
Scanning /DIR/test.rb
Loading cache from /Users/user/.cache/rubocop_cache/e0d642490b17155fb6d3d027de6bfcc19fea0dab/6d7a3b621ca1730e04accd938619e4bdab66cfb1/f63835cee0c0fe552762330b4463568dd634b32b
C

Offenses:

test.rb:4:1: C: [Correctable] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
if super_long_variable_name == 'my_other_super_long_string'
^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
Finished in 0.14378399995621294 seconds
```

## Steps to reproduce the problem

1. Create a new file
2. Write an if condition with a long condition. 
3. Then, raise an error in the condition body with a long message

## RuboCop version

```
1.13.0 (using Parser 3.0.1.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-darwin20)
```

**Comment by koic** [comment]
I couldn't reproduce this issue. Is `Max: 90` set in `Layout/LineLength` of .rubocop.yml?

```ruby
% cat example.rb
# test.rb
super_long_variable_name = 12
if super_long_variable_name == 'my_other_super_long_string'
  raise 'this is my super long error message'
end
```

```yaml
% cat .rubocop.yml
Layout/LineLength:
  Max: 90
```

```console
% bundle exec rubocop -a --only Style/IfUnlessModifier
(snip)

Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by Hansenq** [comment]
Hey, yes it is. Reproduced my entire .rubocop.yml below. If it's only an issue I'm seeing, I can continue trying to figure it out, but any pointers/tips would be appreciated!

```yml
inherit_gem:
  prettier: rubocop.yml

AllCops:
  NewCops: enable
  TargetRubyVersion: 3.0.0
  Exclude:
    - 'node_modules/**/*'
    - 'tmp/**/*'
    - 'vendor/**/*'
    - '.git/**/*'

Layout/LineLength:
  AutoCorrect: true
  Max: 90

Layout/SpaceInsideHashLiteralBraces:
  EnforcedStyle: no_space

Layout/SpaceInsideBlockBraces:
  EnforcedStyle: no_space
  SpaceBeforeBlockParameters: false

Metrics/AbcSize:
  Max: 25

Metrics/ClassLength:
  Max: 500

Metrics/MethodLength:
  Max: 50

Metrics/ModuleLength:
  Max: 500

Style/Documentation:
  Enabled: false

Style/FrozenStringLiteralComment:
  EnforcedStyle: always_true

Style/NegatedIf:
  Enabled: false

Style/PercentLiteralDelimiters:
  PreferredDelimiters:
    default: '()'

Style/SymbolArray:
  EnforcedStyle: brackets

Style/TrailingCommaInArrayLiteral:
  EnforcedStyleForMultiline: consistent_comma

Style/TrailingCommaInArguments:
  EnforcedStyleForMultiline: consistent_comma

Style/TrailingCommaInHashLiteral:
  EnforcedStyleForMultiline: consistent_comma
```

**Comment by Hansenq** [comment]
Ah, it looks like the prettier's rubocop.yml is interfering with this specific style, for some reason. Sorry to both y'all then; thanks for looking into this!

**Comment by Hansenq** [comment]
Ah, just to close the loop on this, looks like the prettier-ruby [rubocop.yml](https://github.com/prettier/plugin-ruby/blob/master/rubocop.yml) file disables all `Layout/*` cops, which means that `Layout/LineLength` is disabled, so `Style/IfUnlessModifier` thinks that line lengths are unlimited. I'll go open a request in their library!
